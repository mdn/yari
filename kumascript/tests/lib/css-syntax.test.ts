import Environment, { KumaThis } from "../../src/environment.js";
import { getCSSSyntax } from "../../src/lib/css-syntax.js";

async function render(slug: string, pageType: string) {
  const environment = new Environment({
    locale: "en-US",
    slug,
    "page-type": pageType,
  });
  const context = environment.getExecutionContext([]) as KumaThis;
  return await getCSSSyntax(context);
}

describe("CSSSyntax", () => {
  it("renders at-rule", async () => {
    expect(await render("Web/CSS/@import", "css-at-rule")).toMatchSnapshot(
      "@import"
    );
  });

  it("renders at-rule-descriptor", async () => {
    expect(
      await render("Web/CSS/@font-face/src", "css-at-rule-descriptor")
    ).toMatchSnapshot("@font-face/src");
  });

  it("renders function", async () => {
    expect(
      await render("Web/CSS/basic-shape/polygon", "css-function")
    ).toMatchSnapshot("polygon");
    expect(await render("Web/CSS/sin", "css-function")).toMatchSnapshot("sin");
  });

  it("renders property", async () => {
    expect(await render("Web/CSS/box-shadow", "css-property")).toMatchSnapshot(
      "box-shadow"
    );
  });

  it("renders shorthand-property", async () => {
    expect(
      await render("Web/CSS/overflow", "css-shorthand-property")
    ).toMatchSnapshot("overflow");
  });

  it("renders type", async () => {
    expect(await render("Web/CSS/ratio", "css-type")).toMatchSnapshot("ratio");
    expect(await render("Web/CSS/alpha-value", "css-type")).toMatchSnapshot(
      "alpha-value"
    );
    // Note that Web/CSS/content-replacement doesn't really exist.
    expect(
      await render("Web/CSS/content-replacement", "css-type")
    ).toMatchSnapshot("content-replacement");
  });
});
