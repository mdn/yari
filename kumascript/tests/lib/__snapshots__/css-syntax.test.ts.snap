// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CSSSyntax renders at-rule: @import 1`] = `"<pre><span class="token property" id="@import">@import = </span><br/>  @import <a href="/en-US/docs/Web/CSS/Value_definition_syntax#brackets" title="Brackets: enclose several entities, combinators, and multipliers to transform them as a single component">[</a> <span class="token property">&lt;url&gt;</span> <a href="/en-US/docs/Web/CSS/Value_definition_syntax#single_bar" title="Single bar: exactly one of the entities must be present">|</a> <a href="/en-US/docs/Web/CSS/string"><span class="token property">&lt;string&gt;</span></a> <a href="/en-US/docs/Web/CSS/Value_definition_syntax#brackets" title="Brackets: enclose several entities, combinators, and multipliers to transform them as a single component">]</a> <a href="/en-US/docs/Web/CSS/Value_definition_syntax#brackets" title="Brackets: enclose several entities, combinators, and multipliers to transform them as a single component">[</a> <span class="token keyword">layer</span> <a href="/en-US/docs/Web/CSS/Value_definition_syntax#single_bar" title="Single bar: exactly one of the entities must be present">|</a> <span class="token function">layer(</span> <span class="token property">&lt;layer-name&gt;</span> <span class="token function">)</span> <a href="/en-US/docs/Web/CSS/Value_definition_syntax#brackets" title="Brackets: enclose several entities, combinators, and multipliers to transform them as a single component">]</a><a href="/en-US/docs/Web/CSS/Value_definition_syntax#question_mark" title="Question mark: the entity is optional">?</a> <span class="token property">&lt;import-conditions&gt;</span> ;  <br/><br/><span class="token property" id="&lt;url&gt;">&lt;url&gt; = </span><br/>  <span class="token property">&lt;url()&gt;</span>  <a href="/en-US/docs/Web/CSS/Value_definition_syntax#single_bar" title="Single bar: exactly one of the entities must be present">|</a><br/>  <span class="token property">&lt;src()&gt;</span>  <br/><br/><span class="token property" id="&lt;layer-name&gt;">&lt;layer-name&gt; = </span><br/>  <a href="/en-US/docs/Web/CSS/ident"><span class="token property">&lt;ident&gt;</span></a> <a href="/en-US/docs/Web/CSS/Value_definition_syntax#brackets" title="Brackets: enclose several entities, combinators, and multipliers to transform them as a single component">[</a> '.' <a href="/en-US/docs/Web/CSS/ident"><span class="token property">&lt;ident&gt;</span></a> <a href="/en-US/docs/Web/CSS/Value_definition_syntax#brackets" title="Brackets: enclose several entities, combinators, and multipliers to transform them as a single component">]</a><a href="/en-US/docs/Web/CSS/Value_definition_syntax#asterisk" title="Asterisk: the entity may occur zero, one or several times">*</a>  <br/><br/><span class="token property" id="&lt;import-conditions&gt;">&lt;import-conditions&gt; = </span><br/>  <a href="/en-US/docs/Web/CSS/Value_definition_syntax#brackets" title="Brackets: enclose several entities, combinators, and multipliers to transform them as a single component">[</a> <span class="token function">supports(</span> <a href="/en-US/docs/Web/CSS/Value_definition_syntax#brackets" title="Brackets: enclose several entities, combinators, and multipliers to transform them as a single component">[</a> <span class="token property">&lt;supports-condition&gt;</span> <a href="/en-US/docs/Web/CSS/Value_definition_syntax#single_bar" title="Single bar: exactly one of the entities must be present">|</a> <a href="/en-US/docs/Web/CSS/declaration"><span class="token property">&lt;declaration&gt;</span></a> <a href="/en-US/docs/Web/CSS/Value_definition_syntax#brackets" title="Brackets: enclose several entities, combinators, and multipliers to transform them as a single component">]</a> <span class="token function">)</span> <a href="/en-US/docs/Web/CSS/Value_definition_syntax#brackets" title="Brackets: enclose several entities, combinators, and multipliers to transform them as a single component">]</a><a href="/en-US/docs/Web/CSS/Value_definition_syntax#question_mark" title="Question mark: the entity is optional">?</a> <a href="/en-US/docs/Web/CSS/media-query-list"><span class="token property">&lt;media-query-list&gt;</span></a><a href="/en-US/docs/Web/CSS/Value_definition_syntax#question_mark" title="Question mark: the entity is optional">?</a>  <br/><br/><span class="token property" id="&lt;url()&gt;">&lt;url()&gt; = </span><br/>  <span class="token function">url(</span> <a href="/en-US/docs/Web/CSS/string"><span class="token property">&lt;string&gt;</span></a> <a href="/en-US/docs/Web/CSS/url-modifier"><span class="token property">&lt;url-modifier&gt;</span></a><a href="/en-US/docs/Web/CSS/Value_definition_syntax#asterisk" title="Asterisk: the entity may occur zero, one or several times">*</a> <span class="token function">)</span>  <a href="/en-US/docs/Web/CSS/Value_definition_syntax#single_bar" title="Single bar: exactly one of the entities must be present">|</a><br/>  <a href="/en-US/docs/Web/CSS/url-token"><span class="token property">&lt;url-token&gt;</span></a>                      <br/><br/><span class="token property" id="&lt;src()&gt;">&lt;src()&gt; = </span><br/>  <span class="token function">src(</span> <a href="/en-US/docs/Web/CSS/string"><span class="token property">&lt;string&gt;</span></a> <a href="/en-US/docs/Web/CSS/url-modifier"><span class="token property">&lt;url-modifier&gt;</span></a><a href="/en-US/docs/Web/CSS/Value_definition_syntax#asterisk" title="Asterisk: the entity may occur zero, one or several times">*</a> <span class="token function">)</span>  <br/><br/><span class="token property" id="&lt;supports-condition&gt;">&lt;supports-condition&gt; = </span><br/>  <span class="token keyword">not</span> <span class="token property">&lt;supports-in-parens&gt;</span>                            <a href="/en-US/docs/Web/CSS/Value_definition_syntax#single_bar" title="Single bar: exactly one of the entities must be present">|</a><br/>  <span class="token property">&lt;supports-in-parens&gt;</span> <a href="/en-US/docs/Web/CSS/Value_definition_syntax#brackets" title="Brackets: enclose several entities, combinators, and multipliers to transform them as a single component">[</a> <span class="token keyword">and</span> <span class="token property">&lt;supports-in-parens&gt;</span> <a href="/en-US/docs/Web/CSS/Value_definition_syntax#brackets" title="Brackets: enclose several entities, combinators, and multipliers to transform them as a single component">]</a><a href="/en-US/docs/Web/CSS/Value_definition_syntax#asterisk" title="Asterisk: the entity may occur zero, one or several times">*</a>  <a href="/en-US/docs/Web/CSS/Value_definition_syntax#single_bar" title="Single bar: exactly one of the entities must be present">|</a><br/>  <span class="token property">&lt;supports-in-parens&gt;</span> <a href="/en-US/docs/Web/CSS/Value_definition_syntax#brackets" title="Brackets: enclose several entities, combinators, and multipliers to transform them as a single component">[</a> <span class="token keyword">or</span> <span class="token property">&lt;supports-in-parens&gt;</span> <a href="/en-US/docs/Web/CSS/Value_definition_syntax#brackets" title="Brackets: enclose several entities, combinators, and multipliers to transform them as a single component">]</a><a href="/en-US/docs/Web/CSS/Value_definition_syntax#asterisk" title="Asterisk: the entity may occur zero, one or several times">*</a>   <br/><br/><span class="token property" id="&lt;supports-in-parens&gt;">&lt;supports-in-parens&gt; = </span><br/>  ( <span class="token property">&lt;supports-condition&gt;</span> <span class="token function">)</span>  <a href="/en-US/docs/Web/CSS/Value_definition_syntax#single_bar" title="Single bar: exactly one of the entities must be present">|</a><br/>  <span class="token property">&lt;supports-feature&gt;</span>        <a href="/en-US/docs/Web/CSS/Value_definition_syntax#single_bar" title="Single bar: exactly one of the entities must be present">|</a><br/>  <span class="token property">&lt;general-enclosed&gt;</span>        <br/><br/><span class="token property" id="&lt;supports-feature&gt;">&lt;supports-feature&gt; = </span><br/>  <span class="token property">&lt;supports-decl&gt;</span>  <br/><br/><span class="token property" id="&lt;general-enclosed&gt;">&lt;general-enclosed&gt; = </span><br/>  <a href="/en-US/docs/Web/CSS/Value_definition_syntax#brackets" title="Brackets: enclose several entities, combinators, and multipliers to transform them as a single component">[</a> <a href="/en-US/docs/Web/CSS/function-token"><span class="token property">&lt;function-token&gt;</span></a> <a href="/en-US/docs/Web/CSS/any-value"><span class="token property">&lt;any-value&gt;</span></a><a href="/en-US/docs/Web/CSS/Value_definition_syntax#question_mark" title="Question mark: the entity is optional">?</a> <span class="token function">)</span> <a href="/en-US/docs/Web/CSS/Value_definition_syntax#brackets" title="Brackets: enclose several entities, combinators, and multipliers to transform them as a single component">]</a>  <a href="/en-US/docs/Web/CSS/Value_definition_syntax#single_bar" title="Single bar: exactly one of the entities must be present">|</a><br/>  <a href="/en-US/docs/Web/CSS/Value_definition_syntax#brackets" title="Brackets: enclose several entities, combinators, and multipliers to transform them as a single component">[</a> ( <a href="/en-US/docs/Web/CSS/any-value"><span class="token property">&lt;any-value&gt;</span></a><a href="/en-US/docs/Web/CSS/Value_definition_syntax#question_mark" title="Question mark: the entity is optional">?</a> <span class="token function">)</span> <a href="/en-US/docs/Web/CSS/Value_definition_syntax#brackets" title="Brackets: enclose several entities, combinators, and multipliers to transform them as a single component">]</a>                 <br/><br/><span class="token property" id="&lt;supports-decl&gt;">&lt;supports-decl&gt; = </span><br/>  ( <a href="/en-US/docs/Web/CSS/declaration"><span class="token property">&lt;declaration&gt;</span></a> <span class="token function">)</span>  <br/><br/></pre>"`;

exports[`CSSSyntax renders at-rule-descriptor: @font-face/src 1`] = `"<pre><span class="token property" id="src">src = </span><br/>  <a href="/en-US/docs/Web/CSS/font-src-list"><span class="token property">&lt;font-src-list&gt;</span></a>  <br/><br/></pre>"`;

exports[`CSSSyntax renders function: polygon 1`] = `"<pre><span class="token property" id="&lt;polygon()&gt;">&lt;polygon()&gt; = </span><br/>  <span class="token function">polygon(</span> <a href="/en-US/docs/Web/CSS/fill-rule"><span class="token property">&lt;'fill-rule'&gt;</span></a><a href="/en-US/docs/Web/CSS/Value_definition_syntax#question_mark" title="Question mark: the entity is optional">?</a> <a href="/en-US/docs/Web/CSS/Value_definition_syntax#brackets" title="Brackets: enclose several entities, combinators, and multipliers to transform them as a single component">[</a> <span class="token keyword">round</span> <a href="/en-US/docs/Web/CSS/length"><span class="token property">&lt;length&gt;</span></a> <a href="/en-US/docs/Web/CSS/Value_definition_syntax#brackets" title="Brackets: enclose several entities, combinators, and multipliers to transform them as a single component">]</a><a href="/en-US/docs/Web/CSS/Value_definition_syntax#question_mark" title="Question mark: the entity is optional">?</a> , <a href="/en-US/docs/Web/CSS/Value_definition_syntax#brackets" title="Brackets: enclose several entities, combinators, and multipliers to transform them as a single component">[</a> <span class="token property">&lt;length-percentage&gt;</span> <span class="token property">&lt;length-percentage&gt;</span> <a href="/en-US/docs/Web/CSS/Value_definition_syntax#brackets" title="Brackets: enclose several entities, combinators, and multipliers to transform them as a single component">]</a><a href="/en-US/docs/Web/CSS/Value_definition_syntax#hash_mark" title="Hash mark: the entity is repeated one or several times, each occurence separated by a comma">#</a> <span class="token function">)</span>  <br/><br/><span class="token property" id="&lt;fill-rule&gt;">&lt;fill-rule&gt; = </span><br/>  <span class="token keyword">nonzero</span>  <a href="/en-US/docs/Web/CSS/Value_definition_syntax#single_bar" title="Single bar: exactly one of the entities must be present">|</a><br/>  <span class="token keyword">evenodd</span>  <br/><br/><span class="token property" id="&lt;length-percentage&gt;">&lt;length-percentage&gt; = </span><br/>  <a href="/en-US/docs/Web/CSS/length"><span class="token property">&lt;length&gt;</span></a>      <a href="/en-US/docs/Web/CSS/Value_definition_syntax#single_bar" title="Single bar: exactly one of the entities must be present">|</a><br/>  <a href="/en-US/docs/Web/CSS/percentage"><span class="token property">&lt;percentage&gt;</span></a>  <br/><br/></pre>"`;

exports[`CSSSyntax renders function: sin 1`] = `"<pre><span class="token property" id="&lt;sin()&gt;">&lt;sin()&gt; = </span><br/>  <span class="token function">sin(</span> <span class="token property">&lt;calc-sum&gt;</span> <span class="token function">)</span>  <br/><br/><span class="token property" id="&lt;calc-sum&gt;">&lt;calc-sum&gt; = </span><br/>  <span class="token property">&lt;calc-product&gt;</span> <a href="/en-US/docs/Web/CSS/Value_definition_syntax#brackets" title="Brackets: enclose several entities, combinators, and multipliers to transform them as a single component">[</a> <a href="/en-US/docs/Web/CSS/Value_definition_syntax#brackets" title="Brackets: enclose several entities, combinators, and multipliers to transform them as a single component">[</a> '+' <a href="/en-US/docs/Web/CSS/Value_definition_syntax#single_bar" title="Single bar: exactly one of the entities must be present">|</a> '-' <a href="/en-US/docs/Web/CSS/Value_definition_syntax#brackets" title="Brackets: enclose several entities, combinators, and multipliers to transform them as a single component">]</a> <span class="token property">&lt;calc-product&gt;</span> <a href="/en-US/docs/Web/CSS/Value_definition_syntax#brackets" title="Brackets: enclose several entities, combinators, and multipliers to transform them as a single component">]</a><a href="/en-US/docs/Web/CSS/Value_definition_syntax#asterisk" title="Asterisk: the entity may occur zero, one or several times">*</a>  <br/><br/><span class="token property" id="&lt;calc-product&gt;">&lt;calc-product&gt; = </span><br/>  <span class="token property">&lt;calc-value&gt;</span> <a href="/en-US/docs/Web/CSS/Value_definition_syntax#brackets" title="Brackets: enclose several entities, combinators, and multipliers to transform them as a single component">[</a> <a href="/en-US/docs/Web/CSS/Value_definition_syntax#brackets" title="Brackets: enclose several entities, combinators, and multipliers to transform them as a single component">[</a> '*' <a href="/en-US/docs/Web/CSS/Value_definition_syntax#single_bar" title="Single bar: exactly one of the entities must be present">|</a> '/' <a href="/en-US/docs/Web/CSS/Value_definition_syntax#brackets" title="Brackets: enclose several entities, combinators, and multipliers to transform them as a single component">]</a> <span class="token property">&lt;calc-value&gt;</span> <a href="/en-US/docs/Web/CSS/Value_definition_syntax#brackets" title="Brackets: enclose several entities, combinators, and multipliers to transform them as a single component">]</a><a href="/en-US/docs/Web/CSS/Value_definition_syntax#asterisk" title="Asterisk: the entity may occur zero, one or several times">*</a>  <br/><br/><span class="token property" id="&lt;calc-value&gt;">&lt;calc-value&gt; = </span><br/>  <a href="/en-US/docs/Web/CSS/number"><span class="token property">&lt;number&gt;</span></a>        <a href="/en-US/docs/Web/CSS/Value_definition_syntax#single_bar" title="Single bar: exactly one of the entities must be present">|</a><br/>  <a href="/en-US/docs/Web/CSS/dimension"><span class="token property">&lt;dimension&gt;</span></a>     <a href="/en-US/docs/Web/CSS/Value_definition_syntax#single_bar" title="Single bar: exactly one of the entities must be present">|</a><br/>  <a href="/en-US/docs/Web/CSS/percentage"><span class="token property">&lt;percentage&gt;</span></a>    <a href="/en-US/docs/Web/CSS/Value_definition_syntax#single_bar" title="Single bar: exactly one of the entities must be present">|</a><br/>  <span class="token property">&lt;calc-keyword&gt;</span>  <a href="/en-US/docs/Web/CSS/Value_definition_syntax#single_bar" title="Single bar: exactly one of the entities must be present">|</a><br/>  ( <span class="token property">&lt;calc-sum&gt;</span> <span class="token function">)</span>  <br/><br/><span class="token property" id="&lt;calc-keyword&gt;">&lt;calc-keyword&gt; = </span><br/>  <span class="token keyword">e</span>          <a href="/en-US/docs/Web/CSS/Value_definition_syntax#single_bar" title="Single bar: exactly one of the entities must be present">|</a><br/>  <span class="token keyword">pi</span>         <a href="/en-US/docs/Web/CSS/Value_definition_syntax#single_bar" title="Single bar: exactly one of the entities must be present">|</a><br/>  <span class="token keyword">infinity</span>   <a href="/en-US/docs/Web/CSS/Value_definition_syntax#single_bar" title="Single bar: exactly one of the entities must be present">|</a><br/>  <span class="token keyword">-infinity</span>  <a href="/en-US/docs/Web/CSS/Value_definition_syntax#single_bar" title="Single bar: exactly one of the entities must be present">|</a><br/>  <span class="token keyword">NaN</span>        <br/><br/></pre>"`;

exports[`CSSSyntax renders property: box-shadow 1`] = `"<pre><span class="token property" id="box-shadow">box-shadow = </span><br/>  <span class="token property">&lt;spread-shadow&gt;</span><a href="/en-US/docs/Web/CSS/Value_definition_syntax#hash_mark" title="Hash mark: the entity is repeated one or several times, each occurence separated by a comma">#</a>  <br/><br/><span class="token property" id="&lt;spread-shadow&gt;">&lt;spread-shadow&gt; = </span><br/>  <a href="/en-US/docs/Web/CSS/box-shadow-color"><span class="token property">&lt;'box-shadow-color'&gt;</span></a><a href="/en-US/docs/Web/CSS/Value_definition_syntax#question_mark" title="Question mark: the entity is optional">?</a>                               <a href="/en-US/docs/Web/CSS/Value_definition_syntax#double_ampersand" title="Double ampersand: all of the entities must be present, in any order">&&</a><br/>  <a href="/en-US/docs/Web/CSS/Value_definition_syntax#brackets" title="Brackets: enclose several entities, combinators, and multipliers to transform them as a single component">[</a> <a href="/en-US/docs/Web/CSS/box-shadow-offset"><span class="token property">&lt;'box-shadow-offset'&gt;</span></a> <a href="/en-US/docs/Web/CSS/Value_definition_syntax#brackets" title="Brackets: enclose several entities, combinators, and multipliers to transform them as a single component">[</a> <a href="/en-US/docs/Web/CSS/box-shadow-blur"><span class="token property">&lt;'box-shadow-blur'&gt;</span></a> <a href="/en-US/docs/Web/CSS/box-shadow-spread"><span class="token property">&lt;'box-shadow-spread'&gt;</span></a><a href="/en-US/docs/Web/CSS/Value_definition_syntax#question_mark" title="Question mark: the entity is optional">?</a> <a href="/en-US/docs/Web/CSS/Value_definition_syntax#brackets" title="Brackets: enclose several entities, combinators, and multipliers to transform them as a single component">]</a><a href="/en-US/docs/Web/CSS/Value_definition_syntax#question_mark" title="Question mark: the entity is optional">?</a> <a href="/en-US/docs/Web/CSS/Value_definition_syntax#brackets" title="Brackets: enclose several entities, combinators, and multipliers to transform them as a single component">]</a>  <a href="/en-US/docs/Web/CSS/Value_definition_syntax#double_ampersand" title="Double ampersand: all of the entities must be present, in any order">&&</a><br/>  <a href="/en-US/docs/Web/CSS/box-shadow-position"><span class="token property">&lt;'box-shadow-position'&gt;</span></a><a href="/en-US/docs/Web/CSS/Value_definition_syntax#question_mark" title="Question mark: the entity is optional">?</a>                            <br/><br/><span class="token property" id="&lt;box-shadow-color&gt;">&lt;box-shadow-color&gt; = </span><br/>  <a href="/en-US/docs/Web/CSS/color_value"><span class="token property">&lt;color&gt;</span></a><a href="/en-US/docs/Web/CSS/Value_definition_syntax#hash_mark" title="Hash mark: the entity is repeated one or several times, each occurence separated by a comma">#</a>  <br/><br/><span class="token property" id="&lt;box-shadow-offset&gt;">&lt;box-shadow-offset&gt; = </span><br/>  <a href="/en-US/docs/Web/CSS/Value_definition_syntax#brackets" title="Brackets: enclose several entities, combinators, and multipliers to transform them as a single component">[</a> <span class="token keyword">none</span> <a href="/en-US/docs/Web/CSS/Value_definition_syntax#single_bar" title="Single bar: exactly one of the entities must be present">|</a> <a href="/en-US/docs/Web/CSS/length"><span class="token property">&lt;length&gt;</span></a><a href="/en-US/docs/Web/CSS/Value_definition_syntax#curly_braces" title="Curly braces: encloses two integers defining the minimal and maximal numbers of occurrences of the entity, or a single integer defining the exact number required">{2}</a> <a href="/en-US/docs/Web/CSS/Value_definition_syntax#brackets" title="Brackets: enclose several entities, combinators, and multipliers to transform them as a single component">]</a><a href="/en-US/docs/Web/CSS/Value_definition_syntax#hash_mark" title="Hash mark: the entity is repeated one or several times, each occurence separated by a comma">#</a>  <br/><br/><span class="token property" id="&lt;box-shadow-blur&gt;">&lt;box-shadow-blur&gt; = </span><br/>  <a href="/en-US/docs/Web/CSS/length "><span class="token property">&lt;length [0,∞]&gt;</span></a><a href="/en-US/docs/Web/CSS/Value_definition_syntax#hash_mark" title="Hash mark: the entity is repeated one or several times, each occurence separated by a comma">#</a>  <br/><br/><span class="token property" id="&lt;box-shadow-spread&gt;">&lt;box-shadow-spread&gt; = </span><br/>  <a href="/en-US/docs/Web/CSS/length"><span class="token property">&lt;length&gt;</span></a><a href="/en-US/docs/Web/CSS/Value_definition_syntax#hash_mark" title="Hash mark: the entity is repeated one or several times, each occurence separated by a comma">#</a>  <br/><br/><span class="token property" id="&lt;box-shadow-position&gt;">&lt;box-shadow-position&gt; = </span><br/>  <a href="/en-US/docs/Web/CSS/Value_definition_syntax#brackets" title="Brackets: enclose several entities, combinators, and multipliers to transform them as a single component">[</a> <span class="token keyword">outset</span> <a href="/en-US/docs/Web/CSS/Value_definition_syntax#single_bar" title="Single bar: exactly one of the entities must be present">|</a> <span class="token keyword">inset</span> <a href="/en-US/docs/Web/CSS/Value_definition_syntax#brackets" title="Brackets: enclose several entities, combinators, and multipliers to transform them as a single component">]</a><a href="/en-US/docs/Web/CSS/Value_definition_syntax#hash_mark" title="Hash mark: the entity is repeated one or several times, each occurence separated by a comma">#</a>  <br/><br/></pre>"`;

exports[`CSSSyntax renders shorthand-property: overflow 1`] = `"<pre><span class="token property" id="overflow">overflow = </span><br/>  <a href="/en-US/docs/Web/CSS/overflow-block"><span class="token property">&lt;'overflow-block'&gt;</span></a><a href="/en-US/docs/Web/CSS/Value_definition_syntax#curly_braces" title="Curly braces: encloses two integers defining the minimal and maximal numbers of occurrences of the entity, or a single integer defining the exact number required">{1,2}</a>  <br/><br/><span class="token property" id="&lt;overflow-block&gt;">&lt;overflow-block&gt; = </span><br/>  <span class="token keyword">visible</span>  <a href="/en-US/docs/Web/CSS/Value_definition_syntax#single_bar" title="Single bar: exactly one of the entities must be present">|</a><br/>  <span class="token keyword">hidden</span>   <a href="/en-US/docs/Web/CSS/Value_definition_syntax#single_bar" title="Single bar: exactly one of the entities must be present">|</a><br/>  <span class="token keyword">clip</span>     <a href="/en-US/docs/Web/CSS/Value_definition_syntax#single_bar" title="Single bar: exactly one of the entities must be present">|</a><br/>  <span class="token keyword">scroll</span>   <a href="/en-US/docs/Web/CSS/Value_definition_syntax#single_bar" title="Single bar: exactly one of the entities must be present">|</a><br/>  <span class="token keyword">auto</span>     <br/><br/></pre>"`;

exports[`CSSSyntax renders type: alpha-value 1`] = `"<pre><span class="token property" id="&lt;alpha-value&gt;">&lt;alpha-value&gt; = </span><br/>  <a href="/en-US/docs/Web/CSS/number"><span class="token property">&lt;number&gt;</span></a>      <a href="/en-US/docs/Web/CSS/Value_definition_syntax#single_bar" title="Single bar: exactly one of the entities must be present">|</a><br/>  <a href="/en-US/docs/Web/CSS/percentage"><span class="token property">&lt;percentage&gt;</span></a>  <br/><br/></pre>"`;

exports[`CSSSyntax renders type: content-replacement 1`] = `"<pre><span class="token property" id="&lt;content-replacement&gt;">&lt;content-replacement&gt; = </span><br/>  <span class="token property">&lt;image&gt;</span>  <br/><br/><span class="token property" id="&lt;image&gt;">&lt;image&gt; = </span><br/>  <span class="token property">&lt;url&gt;</span>       <a href="/en-US/docs/Web/CSS/Value_definition_syntax#single_bar" title="Single bar: exactly one of the entities must be present">|</a><br/>  <a href="/en-US/docs/Web/CSS/gradient"><span class="token property">&lt;gradient&gt;</span></a>  <br/><br/><span class="token property" id="&lt;url&gt;">&lt;url&gt; = </span><br/>  <span class="token property">&lt;url()&gt;</span>  <a href="/en-US/docs/Web/CSS/Value_definition_syntax#single_bar" title="Single bar: exactly one of the entities must be present">|</a><br/>  <span class="token property">&lt;src()&gt;</span>  <br/><br/><span class="token property" id="&lt;url()&gt;">&lt;url()&gt; = </span><br/>  <span class="token function">url(</span> <a href="/en-US/docs/Web/CSS/string"><span class="token property">&lt;string&gt;</span></a> <a href="/en-US/docs/Web/CSS/url-modifier"><span class="token property">&lt;url-modifier&gt;</span></a><a href="/en-US/docs/Web/CSS/Value_definition_syntax#asterisk" title="Asterisk: the entity may occur zero, one or several times">*</a> <span class="token function">)</span>  <a href="/en-US/docs/Web/CSS/Value_definition_syntax#single_bar" title="Single bar: exactly one of the entities must be present">|</a><br/>  <a href="/en-US/docs/Web/CSS/url-token"><span class="token property">&lt;url-token&gt;</span></a>                      <br/><br/><span class="token property" id="&lt;src()&gt;">&lt;src()&gt; = </span><br/>  <span class="token function">src(</span> <a href="/en-US/docs/Web/CSS/string"><span class="token property">&lt;string&gt;</span></a> <a href="/en-US/docs/Web/CSS/url-modifier"><span class="token property">&lt;url-modifier&gt;</span></a><a href="/en-US/docs/Web/CSS/Value_definition_syntax#asterisk" title="Asterisk: the entity may occur zero, one or several times">*</a> <span class="token function">)</span>  <br/><br/></pre>"`;

exports[`CSSSyntax renders type: ratio 1`] = `"<pre><span class="token property" id="&lt;ratio&gt;">&lt;ratio&gt; = </span><br/>  <a href="/en-US/docs/Web/CSS/number "><span class="token property">&lt;number [0,∞]&gt;</span></a> <a href="/en-US/docs/Web/CSS/Value_definition_syntax#brackets" title="Brackets: enclose several entities, combinators, and multipliers to transform them as a single component">[</a> / <a href="/en-US/docs/Web/CSS/number "><span class="token property">&lt;number [0,∞]&gt;</span></a> <a href="/en-US/docs/Web/CSS/Value_definition_syntax#brackets" title="Brackets: enclose several entities, combinators, and multipliers to transform them as a single component">]</a><a href="/en-US/docs/Web/CSS/Value_definition_syntax#question_mark" title="Question mark: the entity is optional">?</a>  <br/><br/></pre>"`;
